src/main/java/org/mockito/internal/stubbing/InvocationContainerImpl.java

```diff
@@ -17,24 +12,32 @@
 import org.mockito.invocation.InvocationContainer;
 import org.mockito.invocation.MatchableInvocation;
 import org.mockito.mock.MockCreationSettings;
+import org.mockito.quality.Strictness;
 import org.mockito.stubbing.Answer;
 import org.mockito.stubbing.Stubbing;
 import org.mockito.stubbing.ValidableAnswer;
 
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+
 import static org.mockito.internal.progress.ThreadSafeMockingProgress.mockingProgress;
 
 @SuppressWarnings("unchecked")
 public class InvocationContainerImpl implements InvocationContainer, Serializable {
 
     private static final long serialVersionUID = -5334301962749537177L;
     private final LinkedList<StubbedInvocationMatcher> stubbed = new LinkedList<StubbedInvocationMatcher>();
-    private final List<Answer<?>> answersForStubbing = new ArrayList<Answer<?>>();
+    private final DoAnswerStyleStubbing doAnswerStyleStubbing;
     private final RegisteredInvocations registeredInvocations;
+    private final Strictness mockStrictness;
 
     private MatchableInvocation invocationForStubbing;
 
     public InvocationContainerImpl(MockCreationSettings mockSettings) {
         this.registeredInvocations = createRegisteredInvocations(mockSettings);
+        this.mockStrictness = mockSettings.isLenient() ? Strictness.LENIENT : null;
```

Comment:
We assign `null` here and use it throughout the code. I still think this is brittle, as it is really easy to miss. Moreover, in `addConsecutiveAnswer` `null` is passed in and in `addAnswer` again `null` is propagated. Rather, we should define a default somewhere and use this instead. That way, this field is never `null` and we always know what "not specifying a strictness for this method" actually means (e.g. fallback to `mockStrictness`)

Reply:
Thank you for feedback! I absolutely agree with you on the nulls.

When implementing this feature, I tried to have a default Strictness. This would require us to add new public value to the enum, for example: ```Strictness.UNSPECIFIED```. I decided against it because it's adding a new awkward value to the public API.

Do you think we should add ```Strictness.UNSPECIFIED```? What would that mean to the user? (e.g. how would we document it?).

Reply:
I would add a constant static attribute in `Strictness` like `public static final Strictness DEFAULT = Strictness.LENIENT;` (or whatever value we decide on).

Reply:
For mock objects and stubbings, the default Strictness is "unspecified", e.g. it will be inherited from MockitoSession or JUnit rule. The question is whether we want to add ```Strictness.UNSPECIFIED``` to the public API?

Reply:
I think we should extract this discussion into an issue, as it seems that this is a more fundamental issue with `Strictness`. Since we are using it more and more and it is configurable in multiple places, maybe we should extract all this logic into a separate class? In this way, you would make a chain of configurations, where we take the "most local" definition. Anyways, let's create an issue with the requirements so that we can tinker.

Reply:
There is already an issue: #792 - please write comment there with your idea! What do you mean about extracting it into a separate class?

Reply:
@mockitoguy Filed #1305 of what I meant and to hopefully address this issue.

Reply:
I have been trying to incorporate #1305 in this PR, but sadly `Strictness` is used in many places in the codebase. I feel like the Strictness feature as a whole has had more impact on our overall architecture than I would have liked. For now I am okay with these changes, but I think that we should revisit the whole Strictness story in a later stadium to see if we can reduce duplication and boilerplate code (such as the `LenientStubber`)

Reply:
I agree. Thank you for giving it a shot!

I'll fix the conflicts so that we're ready to merge.